lib/fs/MiqFsError.rb

```diff
@@ -0,0 +1,11 @@
+class MiqFsError < RuntimeError
+end
+
+class DirectoryNotFound < MiqFsError
```

Comment:
@movitto 
I'm concerned about having a generic class name like this defined in the global namespace. Normally, I'd suggest we define these classes under a namespace like `MiqFs::DirectoryNotFound` and maybe even `MiqFS::Error` as well. But since most of this will be changing going forward, maybe we should just make the name less generic, like `MiqFsDirectoryNotFound`.